code/core/write_sdl_ntuple.cc

```diff
@@ -297,10 +314,80 @@ void setOptionalOutputBranches(SDL::Event<SDL::Acc>* event)
     setPixelQuintupletOutputBranches(event);
     setQuintupletOutputBranches(event);
     setPixelTripletOutputBranches(event);
+    setOccupancyBranches(event);
 
 #endif
 }
 
+//________________________________________________________________________________________________________________________________
+void setOccupancyBranches(SDL::Event<SDL::Acc>* event)
+{
+    SDL::trackCandidates& trackCandidatesInGPU = (*event->getTrackCandidates());
+    SDL::triplets& tripletsInGPU = (*event->getTriplets());
+    SDL::segments& segmentsInGPU = (*event->getSegments());
+    SDL::miniDoublets& mdsInGPU = (*event->getMiniDoublets());
+    //SDL::hits& hitsInGPU = (*event->getHits());
+    SDL::modules& modulesInGPU = (*event->getModules());
+    SDL::quintuplets&  quintupletsInGPU = (*event->getQuintuplets());
+    SDL::pixelQuintuplets& pixelQuintupletsInGPU = (*event->getPixelQuintuplets());
+    SDL::pixelTriplets& pixelTripletsInGPU = (*event->getPixelTriplets());
+    //get the occupancies from these dudes
+    std::vector<int> moduleLayer;
+    std::vector<int> moduleSubdet;
+    std::vector<int> moduleRing;
+    std::vector<int> moduleRod;
+    std::vector<int> moduleModule;
+    std::vector<float> moduleEta;
+    std::vector<float> moduleR;
+    std::vector<bool> moduleIsTilted;
+    std::vector<int> trackCandidateOccupancy;
+    std::vector<int> tripletOccupancy;
+    std::vector<int> segmentOccupancy;
+    std::vector<int> mdOccupancy;
+    std::vector<int> quintupletOccupancy;
+
+    bool isTilted;
+
+    for(unsigned int idx = 0; idx <= *(modulesInGPU.nLowerModules); idx++)
+    {
+        //layer = 0, subdet = 0 => pixel module
+        //module, md and segment - need some gymnastics
+        unsigned int lowerIdx = idx;//modulesInGPU.lowerModuleIndices[idx];
+        moduleLayer.push_back(modulesInGPU.layers[lowerIdx]);
+        moduleSubdet.push_back(modulesInGPU.subdets[lowerIdx]);
+        moduleRing.push_back(modulesInGPU.rings[lowerIdx]);
+        moduleRod.push_back(modulesInGPU.rods[lowerIdx]);
+        moduleEta.push_back(modulesInGPU.eta[lowerIdx]);
+        moduleR.push_back(modulesInGPU.r[lowerIdx]);
+        isTilted = (modulesInGPU.subdets[lowerIdx] == 5 and modulesInGPU.sides[lowerIdx] != 3);
+        moduleIsTilted.push_back(isTilted);
+        moduleModule.push_back(modulesInGPU.modules[lowerIdx]);
+        segmentOccupancy.push_back(segmentsInGPU.totOccupancySegments[lowerIdx]);
+        mdOccupancy.push_back(mdsInGPU.totOccupancyMDs[lowerIdx]);
+
+        if(idx < *(modulesInGPU.nLowerModules))
```

Comment:
@YonsiG or @slava77 Do you know why the t5 and t3 occupancies are indexed from 0 to nLowerModules-1 and the md and segment occupancies are indexed from 0 to nLowerModules?

Reply:
I decided to get rid of this, I don't think it will matter when choosing the occupancies.